src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -26,12 +29,14 @@ internal class OAuth2Client
         private readonly Dictionary<string, string> _queryParameters = new Dictionary<string, string>();
         private readonly IHttpManager _httpManager;
         private readonly ITelemetryManager _telemetryManager;
+        private readonly IDeviceAuthManager _deviceAuthManager;
 
-        public OAuth2Client(ICoreLogger logger, IHttpManager httpManager, ITelemetryManager telemetryManager)
+        public OAuth2Client(ICoreLogger logger, IHttpManager httpManager, ITelemetryManager telemetryManager, IDeviceAuthManager deviceAuthManager)
```

Comment:
This code is quite old and needs to be rebased. There is now a `TokenClient` higher-level class that is responsible for the communication with /token endpoint. Not sure if you saw that. 

It's not clear to me if this PKAuth challenge only comes from the /token endpoint or from all the endpoints? If it's only from the /token endpoint, then probably it's better to move the logic to `TokenClient` class.